Add images user journey (#2991)
viamrobotics/docs: docs/use-cases/image-data.md

---
title: "How to capture, filter, and sync image data"
linkTitle: "Capture and sync image data"
weight: 20
type: "docs"
tags: ["data management", "data", "services"]
images: ["/services/ml/collect.svg"]
description: "Capture images from a camera on your machine and selectively sync images to the cloud with filtering."
---

You can use Viam's built-in data management service to capture images from a camera on your machine and sync the images to the cloud.

If you want to capture only certain images, such as those containing a person, you can use a "filtering camera" to selectively capture images based on a computer vision model.

With your images synced to the cloud, you can view images from all your machines in one Viam app interface.
From there, you can use your image data to do things like [train ML models](/use-cases/deploy-ml/).

> <div style="border:1px solid; box-shadow:5px 5px 0 0;padding:1rem; margin:1rem; background-color:#f5f5f5">
> 
> **In this page:**
> 
> 1. [Collect image data and sync it to the cloud](#collect-image-data-and-sync-it-to-the-cloud)
> 2. [Use filtering to collect and sync only certain images](#use-filtering-to-collect-and-sync-only-certain-images)
> 
> </div>

Reviewer: ```suggestion
{{< alert title="In this page" color="tip" >}}

1. [Collect image data and sync it to the cloud](#collect-image-data-and-sync-it-to-the-cloud)
2. [Use filtering to collect and sync only certain images](#use-filtering-to-collect-and-sync-only-certain-images)

{{< /alert >}}
```

## Collect image data and sync it to the cloud

{{< table >}}
{{< tablestep >}}
{{<imgproc src="/icons/components/camera.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="configure a camera component">}}
**1. Configure a camera [_(i)_](/components/camera/)**

> First, [create a machine](/cloud/machines/#add-a-new-machine) if you haven't yet.

Reviewer: Can you add the creating a machine as a prerequisite instead please as with the sensor journeys: https://github.com/viamrobotics/docs/pull/2986/files#commit-suggestions.

{{< /tablestep >}}
{{< tablestep >}}
{{<imgproc src="/services/icons/data-capture.svg" class="fill alignleft" style="max-width: 150px"  declaredimensions=true alt="Train models">}}
**4. Configure data capture and sync on the filtered camera**

Configure data capture and sync just as you would for a webcam.
The filtered camera will only capture image data that passes the filters you configured in the previous step.

Turn off data capture on your webcam if you haven't already, so that you don't capture duplicate or unfiltered images.

{{< /tablestep >}}
{{< tablestep >}}
{{<imgproc src="/services/ml/configure.svg" class="fill alignleft" style="max-width: 150px"  declaredimensions=true alt="Train models">}}
**5. (Optional) Trigger sync with custom logic**

By default, the captured data syncs at the regular interval you specified in the data capture config.
If you need to trigger sync in a different way, see [Trigger cloud sync conditionally](/services/data/trigger-sync/) for a documented example of syncing data only at certain times of day.

{{< /tablestep >}}
{{< /table >}}

## Next steps

> {{< cards >}}
> {{% card link="/use-cases/deploy-ml/" %}}
> {{% card link="/services/data/query/" %}}

Reviewer: ```suggestion
## Next steps

Now that you have collected image data, you can also [train computer vision models](/use-cases/deploy-ml/) or [query your data](/services/data/query/).

{{< cards >}}
{{% card link="/use-cases/deploy-ml/" %}}
{{% card link="/services/data/query/" %}}
{{< /cards >}}

To see image data filtering in action, check out these tutorials:

{{< cards >}}
{{% card link="/tutorials/projects/filtered-camera/" %}}
{{% card link="/tutorials/configure/pet-photographer.md" %}}
```

Author: Adding a colon
```suggestion
## Next steps
Now that you have collected image data, you can also [train computer vision models](/use-cases/deploy-ml/) or [query your data](/services/data/query/):

{{< cards >}}
{{% card link="/use-cases/deploy-ml/" %}}
{{% card link="/services/data/query/" %}}
{{< /cards >}}

To see image data filtering in action, check out these tutorials:

{{< cards >}}
{{% card link="/tutorials/projects/filtered-camera/" %}}
{{% card link="/tutorials/configure/pet-photographer.md" %}}
```

{{< alert title="In this page" color="tip" >}}

1. [Collect image data and sync it to the cloud](#collect-image-data-and-sync-it-to-the-cloud)
2. [Use filtering to collect and sync only certain images](#use-filtering-to-collect-and-sync-only-certain-images)

{{< /alert >}}

## Prerequisites

{{% expand "A running machine connected to the Viam app. Click to see instructions." %}}

{{% snippet "setup.md" %}}

{{% /expand%}}

## Collect image data and sync it to the cloud

{{< table >}}
{{< tablestep >}}
{{<imgproc src="/icons/components/camera.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="configure a camera component">}}
**1. Configure a camera [_(i)_](/components/camera/)**

Configure a camera component, such as a [webcam](/components/camera/webcam/), on your machine.

{{< /tablestep >}}
{{< tablestep >}}
{{<imgproc src="/services/icons/data-management.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="Collect data">}}
**2. Enable the data management service [_(i)_](/services/data/)**

In your camera component configuration panel, find the **Data capture** section.

> Click **Add method** and follow the prompt to **Create data management service**.

Author: ok

{{< /tablestep >}}
{{< tablestep >}}
{{<imgproc src="/services/icons/data-capture.svg" class="fill alignleft" style="max-width: 150px"  declaredimensions=true alt="Train models">}}
**4. Configure data capture and sync on the filtered camera**

Configure data capture and sync just as you would for a webcam.
The filtered camera will only capture image data that passes the filters you configured in the previous step.

Turn off data capture on your webcam if you haven't already, so that you don't capture duplicate or unfiltered images.

{{< /tablestep >}}
{{< tablestep >}}
{{<imgproc src="/services/ml/configure.svg" class="fill alignleft" style="max-width: 150px"  declaredimensions=true alt="Train models">}}
**5. (Optional) Trigger sync with custom logic**

By default, the captured data syncs at the regular interval you specified in the data capture config.
If you need to trigger sync in a different way, see [Trigger cloud sync conditionally](/services/data/trigger-sync/) for a documented example of syncing data only at certain times of day.

{{< /tablestep >}}
{{< /table >}}

## Next steps

## Next steps

> Now that you have collected image data, you can also [train computer vision models](/use-cases/deploy-ml/) or [query your data](/services/data/query/):

Author: Okay, applying. Does the CLI count as "programmatically" even though it's not in a script as such?
